pymodbus/server/async_io.py

```diff
@@ -201,8 +201,6 @@ async def handle(self):  # pylint: disable=too-complex
                 else:
                     addr = (None,)  # empty tuple
 
-                if not isinstance(slaves, (list, tuple)):
```

Comment:
This is still open, and should not be removed.

Reply:
You understand the code better than me, but I'm still confused how `handle.slaves` will ever be an integer.

At the highest level creating an async server uses the `slaves` argument to instantiate either a `ModbusSlaveContext`, or a dict of them.
https://github.com/pymodbus-dev/pymodbus/blob/f1a6f925790cdd7353b49f55e7112a93234d827a/examples/server_async.py#L82-L120

Then it creates a `ModbusServerContext` passing `context` (not an integer).
https://github.com/pymodbus-dev/pymodbus/blob/f1a6f925790cdd7353b49f55e7112a93234d827a/examples/server_async.py#L122-L123

`ModbusServerContext` has `slaves` only as a dict (not an integer).
https://github.com/pymodbus-dev/pymodbus/blob/f1a6f925790cdd7353b49f55e7112a93234d827a/pymodbus/datastore/context.py#L119-L137

Reply:
I tried 3 things:
(1) running the test suite (it passes)
(2) changing to
```python
if not isinstance(slaves, (list, tuple)):
  raise NotImplementedError
```
and running the test suite (it passes).
(3) Using mypy by changing `[mypy] check_untyped_defs = true` in `setup.cfg` and using `reveal_type(slave)` (which wasn't useful, there aren't enough type definitions so it says `Any`).

I can't think of any further tests, since I only ever use the client.